cairo/ethereum/cancun/vm/instructions.cairo

```diff
@@ -0,0 +1,703 @@
+    jmp end;
+    call unknown_opcode;  // 0xd3
+    jmp end;
+    call unknown_opcode;  // 0xd4
+    jmp end;
+    call unknown_opcode;  // 0xd5
+    jmp end;
+    call unknown_opcode;  // 0xd6
+    jmp end;
+    call unknown_opcode;  // 0xd7
+    jmp end;
+    call unknown_opcode;  // 0xd8
+    jmp end;
+    call unknown_opcode;  // 0xd9
+    jmp end;
+    call unknown_opcode;  // 0xda
+    jmp end;
+    call unknown_opcode;  // 0xdb
+    jmp end;
+    call unknown_opcode;  // 0xdc
+    jmp end;
+    call unknown_opcode;  // 0xdd
+    jmp end;
+    call unknown_opcode;  // 0xde
+    jmp end;
+    call unknown_opcode;  // 0xdf
+    jmp end;
+    call unknown_opcode;  // 0xe0
+    jmp end;
+    call unknown_opcode;  // 0xe1
+    jmp end;
+    call unknown_opcode;  // 0xe2
+    jmp end;
+    call unknown_opcode;  // 0xe3
+    jmp end;
+    call unknown_opcode;  // 0xe4
+    jmp end;
+    call unknown_opcode;  // 0xe5
+    jmp end;
+    call unknown_opcode;  // 0xe6
+    jmp end;
+    call unknown_opcode;  // 0xe7
+    jmp end;
+    call unknown_opcode;  // 0xe8
+    jmp end;
+    call unknown_opcode;  // 0xe9
+    jmp end;
+    call unknown_opcode;  // 0xea
+    jmp end;
+    call unknown_opcode;  // 0xeb
+    jmp end;
+    call unknown_opcode;  // 0xec
+    jmp end;
+    call unknown_opcode;  // 0xed
+    jmp end;
+    call unknown_opcode;  // 0xee
+    jmp end;
+    call unknown_opcode;  // 0xef
+    jmp end;
+    call unknown_opcode;  // 0xf0 //TODO: create
+    jmp end;
+    call unknown_opcode;  // 0xf1 //TODO: call
+    jmp end;
+    call unknown_opcode;  // 0xf2 //TODO: callcode
+    jmp end;
+    call unknown_opcode;  // 0xf3 //TODO: return
+    jmp end;
+    call unknown_opcode;  // 0xf4 //TODO: delegatecall
+    jmp end;
+    call unknown_opcode;  // 0xf5 //TODO: create2
+    jmp end;
+    call unknown_opcode;  // 0xf6
+    jmp end;
+    call unknown_opcode;  // 0xf7
+    jmp end;
+    call unknown_opcode;  // 0xf8
+    jmp end;
+    call unknown_opcode;  // 0xf9
+    jmp end;
+    call unknown_opcode;  // 0xfa: TODO: staticcall
+    jmp end;
+    call unknown_opcode;  // 0xfb
+    jmp end;
+    call unknown_opcode;  // 0xfc
+    jmp end;
+    call revert;  // 0xfd
+    jmp end;
+    call unknown_opcode;  // 0xfe
+    jmp end;
+    call unknown_opcode;  // 0xff TODO: selfdestruct
+    jmp end;
+
+    end:
+    let range_check_ptr = [ap - 6];
+    let bitwise_ptr = cast([ap - 5], BitwiseBuiltin*);
+    let keccak_ptr = cast([ap - 4], KeccakBuiltin*);
+    let poseidon_ptr = cast([ap - 3], PoseidonBuiltin*);
+    let evm_ = cast([ap - 2], EvmStruct*);
+    let evm = Evm(evm_);
+    let error = cast([ap - 1], EthereumException*);
+
+    return error;
+}
+
+func unknown_opcode{
+    range_check_ptr,
+    bitwise_ptr: BitwiseBuiltin*,
+    keccak_ptr: KeccakBuiltin*,
+    poseidon_ptr: PoseidonBuiltin*,
+    evm: Evm,
+}() {
+    with_attr error_message("ValueError") {
+        assert 1 = 0;
+    }
+    return ();
```

Comment:
Shouldn't it return an invalidOpcode error instead?
In the execute_code function of EELS
It would allow to not handle the ValueError case in execute_code and raise directly the correct error

```
            try:
                op = Ops(evm.code[evm.pc])
            except ValueError:
                raise InvalidOpcode(evm.code[evm.pc])
```

Reply:
yes, good idea